tests/page_test.go

```diff
@@ -1699,3 +1699,59 @@ func TestShadowDOMAndDocumentFragment(t *testing.T) {
 		})
 	}
 }
+
+func TestPageTargetBlank(t *testing.T) {
+	t.Parallel()
+
+	tb := newTestBrowser(t, withHTTPServer())
+	tb.withHandler("/home", func(w http.ResponseWriter, _ *http.Request) {
+		_, err := w.Write([]byte(
+			`<!DOCTYPE html><html><head></head><body>
+				<h1>click please</h1><a href="/link" target="_blank">click me</a>
+			</body></html>`,
+		))
+		require.NoError(t, err)
+	})
+	tb.withHandler("/link", func(w http.ResponseWriter, _ *http.Request) {
+		_, err := w.Write(
+			[]byte(`<!DOCTYPE html><html><head></head><body><h1>you clicked!</h1></body></html>`),
+		)
+		require.NoError(t, err)
+	})
+
+	p := tb.NewPage(nil)
+
+	// Navigate to the page with a link that opens a new page.
+	opts := &common.FrameGotoOptions{
+		Timeout: common.DefaultTimeout,
+	}
+	resp, err := p.Goto(tb.url("/home"), opts)
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+
+	// Current page count should be 1.
+	pp := p.Context().Pages()
+	assert.Equal(t, 1, len(pp))
+
+	// This link should open the link on a new page.
+	err = p.Click("a[href='/link']", common.NewFrameClickOptions(p.Timeout()))
+	require.NoError(t, err)
+
+	// Page count should now be 2.
+	for i := 0; i < 5; i++ {
+		pp = p.Context().Pages()
+		if len(pp) == 2 {
+			break
+		}
+		time.Sleep(time.Millisecond * 500)
+	}
```

Comment:
This test might become a possible flaky test.

Instead of using a loop and `time.Sleep`, what about using:
* `p.Context().WaitForEvent("page", ...)` and,
* `WaitForLoadState` for the newly loaded page?

Reply:
I was avoiding `waitForLoadState` because of https://github.com/grafana/xk6-browser/issues/880, but I'll give that a go 👍 

Nice idea on `WaitForEvent`.

I've added these changes in 8b102b802bc78c412bceb7ec9a3a7739f12502a1

Reply:
👍 Got it. Still, IMO, it's better to fix `waitForLoadState` instead of creating a potentially flaky test.